src/components/course/data/utils.jsx

```diff
@@ -779,3 +779,20 @@ export const getCourseStartDate = ({ contentMetadata, courseRun }) => {
 
   return startDate;
 };
+
+export function useCourseSubjects(course) {
+  const config = getConfig();
+  let subjects = [];
+  let primarySubject = null;
+  if (course?.subjects) {
+    subjects = course.subjects;
+    if (course.subjects.length > 0) {
```

Comment:
nit/suggestion: I wonder if an early return would simplify the nested conditionals? E.g.

```js
function getCourseSubjects(course) {
  if (!course?.subjects?.length) {
    return { subjects: [], primarySubject: null }];
  }
  return {
    subjects: course.subjects,
    primarySubject: {
      ...course.subjects[0],
      url: '...',
    },
  };
}
```

Reply:
Yeah that is much easier to read! Thank you!